ocaml/xapi/xapi_pool_update.ml

```diff
@@ -188,6 +188,51 @@ let guidance_from_string = function
   | "restartXAPI" -> `restartXAPI
   | _ -> raise Bad_update_info
 
+let parse_update_info xml =
+  match xml with
+  | Xml.Element ("update", attr, children) ->
+    let key = List.assoc "key" attr in
+    let uuid = List.assoc "uuid" attr in
+    let name_label = List.assoc "name-label" attr in
+    let installation_size =
+      try
+        Int64.of_string (List.assoc "installation-size" attr)
+      with
+      | _ -> 0L
+    in
+    let guidance = 
+      try
+        match List.assoc "after-apply-guidance" attr with
+        | "" -> []
+        | s ->  List.map guidance_from_string (String.split ' ' s)
+      with
+      | _ -> []
+    in
+    let is_name_description_node = function
+      | Xml.Element ("name-description", _, _) -> true
+      | _ -> false
+    in
+    let name_description = match List.find is_name_description_node children with
+      | Xml.Element("name-description", _, [ Xml.PCData s ]) -> s
+      | _ -> failwith "Malformed or missing <name-description>"
+    in
+    let update_info = {
+      uuid = uuid;
+      name_label = name_label;
+      name_description = name_description;
+      key = key;
+      installation_size = installation_size;
+      after_apply_guidance = guidance;
+    } in
+    update_info
+  | _ -> failwith "Malformed or missing <update>"
```

Comment:
Api_errors are declared here: https://github.com/xapi-project/xen-api/blob/master/ocaml/idl/api_errors.ml, and included in the datamodel here: https://github.com/xapi-project/xen-api/blob/9c23a5a3c03a9ab4fb107c561245a74b7abb4763/ocaml/idl/datamodel.ml#L431-L1350